src/NuGetForUnity/Editor/NugetAssetPostprocessor.cs

```diff
@@ -189,7 +225,29 @@ private static void ModifyImportSettingsOfRoslynAnalyzer([NotNull] PluginImporte
                 plugin.SetExcludeFromAnyPlatform(platform, false);
             }
 
-            AssetDatabase.SetLabels(plugin, new[] { RoslynAnalyzerLabel, ProcessedLabel });
+            var enableRoslynAnalyzer = true;
+
+            // The nuget package can contain analyzers for multiple Roslyn versions.
+            // In that case, for the same package, the most recent version must be chosen out of those available for the current Unity version.
+            var assetPath = Path.GetFullPath(plugin.assetPath);
+            var assetRoslynVersion = GetRoslynVersionNumberFromAnalyzerPath(assetPath);
+            if (assetRoslynVersion != null)
+            {
+                var versionPrefixIndex = assetPath.IndexOf(AnalyzersRoslynVersionsFolderName, StringComparison.Ordinal);
+                var analyzersVersionsRoot = Path.Combine(assetPath.Substring(0, versionPrefixIndex), AnalyzersRoslynVersionsFolderName);
+                var analyzersFolders = Directory.GetDirectories(analyzersVersionsRoot);
+                var enabledRoslynVersions = analyzersFolders.Select(GetRoslynVersionNumberFromAnalyzerPath)
+                    .Where(version => version != null && string.CompareOrdinal(version, MaxSupportedRoslynVersion) <= 0)
```

Comment:
Comparing version number strings using 'string comparison' doesn't work in all cases, e.g. `1.10.0` will be lower than `1.9.0`. You can use the `NugetPackageVersion` class it supports comparison of version numbers including pre-release versions etc.

Reply:
Ok, I fixed it to use NugetPackageVersion.